### ISSUE TITLE
Backup/restore scripts fulls up /tmp FS

### ISSUE THREAD

**Issue by Fixmetal** [jenkinsci/kubernetes-operator#548]
**Describe the bug**
At the moment [backup.sh](https://github.com/jenkinsci/kubernetes-operator/blob/master/backup/pvc/bin/backup.sh#L8) uses  `mktemp`  which uses `/tmp` as default location without any parameters. But `mktemp` doesn't automatically delete what it's created, see [man](https://linux.die.net/man/1/mktemp).
Therefore when having a large Jenkins instance the `/tmp` FS gets all full up quickly. It's an ephemeral FS, but nonetheless, the container seems to get killed because of this. Because of the pod being killed the operator does a full reconcile killing unnecessarily an otherwise perfectly functional Jenkins Master. See logs further down.

Plus: the `exit 0` as the latest command in scripts is unnecessary.

**To Reproduce**
Wait for your backup to be done and check the `/tmp` directory. You will find all the backups done in the past:

```
user@jenkins-master:~$ ls -lRa /tmp
/tmp:
total 0
drwxrwxrwt. 1 root root 72 Apr 21 18:31 .
drwxr-xr-x. 1 root root 60 Apr 21 17:03 ..
drwx------. 2 user user 25 Apr 21 17:04 tmp.2sW66m8cYw
drwx------. 2 user user 25 Apr 21 17:08 tmp.AyvJwaxZhC
drwx------. 2 user user 25 Apr 21 18:29 tmp.lxefb8qzHA
/tmp/tmp.2sW66m8cYw:
total 632264
drwx------. 2 user user        25 Apr 21 17:04 .
drwxrwxrwt. 1 root root        72 Apr 21 18:31 ..
-rw-r--r--. 1 user user 647434750 Apr 21 16:40 3163.tar.gz
/tmp/tmp.AyvJwaxZhC:
total 638660
drwx------. 2 user user        25 Apr 21 17:08 .
drwxrwxrwt. 1 root root        72 Apr 21 18:31 ..
-rw-r--r--. 1 user user 653986615 Apr 21 17:29 3164.tar.gz
/tmp/tmp.lxefb8qzHA:
total 65280
drwx------. 2 user user       25 Apr 21 18:29 .
drwxrwxrwt. 1 root root       72 Apr 21 18:31 ..
-rw-r--r--. 1 user user 41435136 Apr 21 18:31 3165.tar.gz
```

**Additional information**

Kubernetes version: v1.15.3
Jenkins Operator version: v0.5.0

Add error logs about the problem here (operator logs and Kubernetes events).

Logs here:
```
2021-04-21T15:02:13.521Z	INFO	controller-jenkins	base/reconcile.go:370	Container 'backup' is terminated, status '{Name:backup State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:255,Signal:0,Reason:Error,Message:,StartedAt:2021-04-19 19:27:17 +0000 UTC,FinishedAt:2021-04-21 15:02:12 +0000 UTC,ContainerID:docker://26b1bf799490f1d908fe5cbb73631c24be65877c9c7769b31dbc02153ac137c2,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:<our-private-docker-registry>/jenkins-backup:0.1.0-11 ImageID:docker-pullable://<our-private-docker-registry>/jenkins/jenkins-backup@sha256:f68c2c1bef846b4f78dc31f4c122cc9688a786d9dfcdfd6e6949aff046e2c1cc ContainerID:docker://26b1bf799490f1d908fe5cbb73631c24be65877c9c7769b31dbc02153ac137c2 Started:<nil>}'	{"cr": "master"}
[...]
2021-04-21T15:02:46.360Z	WARN	controller-jenkins	jenkins/jenkins_controller.go:136	Reconcile loop failed 10 times with the same error, giving up: pod exec error operation on stream: stdout '' stderr '': unable to upgrade connection: container not found ("backup")	{"cr": "master"}
2021-04-21T15:03:18.991Z	INFO	controller-jenkins	base/pod.go:159	Creating a new Jenkins Master Pod jenkins/jenkins-master	{"cr": "master"}
```

**Comment by Fixmetal** [comment]
@SylwiaBrant : I saw your PR and I think `rm -rf` would be more proper otherwise it wouldn't delete non empty directory

**Comment by KorusMateusz** [comment]
Fixed in PV image v0.1.1
